doc/debugging.rst

```diff
@@ -14,33 +14,91 @@ This allows breakpoints and the execution flow to be viewed and controlled right
+  * Through pressing the configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
+  * By clicking to the left of the line number in an open file in the Editor (adding :kbd:`Shift` for a conditional breakpoint).
+  * With the ``breakpoint()`` function in your code.
+  * Interactively, using the ``!b`` command in an ``ipdb`` session.
+
+GIF: setting conditional breakpoint shift-clicking
 
-* Full GUI control over debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` console commands.
+You can access and edit local and global variables at each breakpoint through the :doc:`variableexplorer`.
 
-.. image:: images/debugging/debugging-condbreakpoint.png
-   :alt: Inset of Spyder's Editor, with a breakpoint set and the condition dialog open
+SCREENSHOT with console and variable explorer showing local and global variables
 
-* Highlighting of the current frame (debugging step) in the :doc:`editor`.
 
-* The ability to access and edit local and global variables at each breakpoint through the :doc:`variableexplorer`, and run many commands in the :doc:`ipythonconsole`.
 
-For a comprehensive but accessible introduction to ``pdb``/``ipdb``, consult Steve Ferg's excellent online guide, `Debugging in Python`_.
+=================
+Advanced features
+=================
+
+You can avoid stepping through other Python packages while debugging by enabling the new :guilabel:`Ignore Python libraries while debugging` option in Spyder's preferences, under :menuselection:`IPython Console --> Debugger --> Debug`.
+This will skip all the built-in and third-party Python modules.
+
+SCREENSHOT with option in preferences
+
+If your code has variables with the same names as Pdb commands (e.g. ``a``, ``step``), you can still refer to them as normal while debugging.
+To call the Pdb command, just add an exclamation point (e.g. ``!a``, ``!step``).
+
+SCREENSHOT with "b" variable
+
+You can have Spyder automatically execute a custom snippet of code every time the debugger stops.
+For example, you can use this to set specific variables, or import commonly-used modules so they are always available while debugging.
+To set this up, go to :menuselection:`Preferences --> IPython Console --> Debugger --> Run code while debugging`, and enter the code that you want to be executed with each step.
+
+SCREENSHOT with Run code while debugging in preferences.
+
+
+
+==================
+Matplotlib support
+==================
+
+Generating Matplotlib figures is fully supported while the debugger is active, including all the different graphics backends.
+Use the ``%matplotlib`` magic to change to an interactive backend (e.g. ``%matplotlib qt5``) to pan, zoom and adjust your plots in a separate window, or switch back to the default ``inline`` (``%matplotlib inline``) to see them displayed right in the console.
```

Comment:
```suggestion
Use the ``%matplotlib`` magic to change to an interactive backend (e.g. ``%matplotlib qt5``) to pan, zoom and adjust your plots in a separate window, or switch back to the default ``inline`` (``%matplotlib inline``) to see them displayed right in the :doc:`plot` pane.
```

* That's the default behavior.

Reply:
Ah, didn't know if that still worked while debugging (the text before led me to believe otherwise).